Add query history to the Preview Web UI

koszti opened the pull request:
## Description

This PR is Work in PROGRESS

Add Query History with search to the preview UI. Partially addressing https://github.com/trinodb/trino/issues/22697

## Additional context and related issues

Retains the same functionality as the existing web UI, enhanced with the updated layout, design, and user experience.

## Screenshots

TODO

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
## Section
* Add query history to the Preview Web UI. ({issue}`24741`)
```

wendigo commented:
I've built this branch but I can only see:

<img width="1702" alt="Screenshot 2025-02-04 at 13 16 31" src="https://github.com/user-attachments/assets/64fe020f-d6a0-44e2-a1d2-6cb64953b33c" />

wendigo commented:
Query text should go up (rather than being in a middle). Or we could stretch the box to be the same height as the details pane on th
<img width="1271" alt="Screenshot 2025-02-04 at 13 26 41" src="https://github.com/user-attachments/assets/80302fb2-bed1-4f7d-95ae-cd0156f6323e" />
e left:

wendigo commented:
This dotted border is looking odd, let's make it solid one:
<img width="378" alt="Screenshot 2025-02-04 at 13 28 07" src="https://github.com/user-attachments/assets/a59e1751-d5ce-4fe4-b72a-4d721edb2cac" />

wendigo commented:
Couple of nits:

Can we make the bottom border in line with the hightlight box?

<img width="673" alt="Screenshot 2025-02-09 at 10 37 40" src="https://github.com/user-attachments/assets/01ca984a-f3a3-4789-a81b-ff768aeac26a" />


I've changed some of the icons, but can we make them smaller and add some right margin to them:

<img width="538" alt="Screenshot 2025-02-09 at 11 07 02" src="https://github.com/user-attachments/assets/02d904ea-a2f1-4c8c-8072-deb6d295534a" />

icons are:

```
import AvTimerIcon from '@mui/icons-material/AvTimer';
import BadgeIcon from '@mui/icons-material/Badge'
import BrokenImageIcon from '@mui/icons-material/BrokenImage';
import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import DevicesIcon from '@mui/icons-material/Devices'
import DownloadingIcon from '@mui/icons-material/Downloading'
import FunctionsIcon from '@mui/icons-material/Functions';
import GraphicEq from '@mui/icons-material/GraphicEq'
import GroupsIcon from '@mui/icons-material/Groups'
import HighlightOff from '@mui/icons-material/HighlightOff'
import HistoryToggleOffIcon from '@mui/icons-material/HistoryToggleOff';
import Memory from '@mui/icons-material/Memory'
import NotStartedIcon from '@mui/icons-material/NotStarted';
import PlayCircleIcon from '@mui/icons-material/PlayCircle';
import QueryBuilderIcon from '@mui/icons-material/QueryBuilder';
```

Can we make query box stand out less? Bring colors a couple of tones down (more grayish)?


a11yDark looks nicer to me:
<img width="1131" alt="Screenshot 2025-02-09 at 11 20 50" src="https://github.com/user-attachments/assets/651df452-903f-4ca1-996d-f34de4a64c23" />

wendigo commented:
@koszti can you rebase this PR? You've added commits instead of rebasing

koszti commented:
@wendigo I addressed all change requests but I messed up this PR with some incorrect rebase and merge. Instead of trying to make the git history and diff clean, I opened another and cleaned PR at https://github.com/trinodb/trino/pull/24972. Can we continue over there?
